common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/file/remote/AbstractRemoteFileSourceProperties.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.module.file.remote;
+
+import java.io.File;
+
+/**
+ * @author David Turanski
+ * @author Gary Russell
+ *
+ */
+public abstract class AbstractRemoteFileSourceProperties {
+
+	/**
+	 * The remote FTP directory, "/" by default.
+	 */
+	private String remoteDir = "/";
+
+	/**
+	 * Set to true to delete remote files after successful transfer, false by default.
```

Comment:
Can't find this comment anymore, maybe you removed it Gary.
To clarify, yes we're using the stuff in META-INF, but that is generated in
99% of the cases (hence my comments about needing proper javadoc, no need
to document the default, etc).
The file you pointed to in my other PR (#44) is needed because
`pathPattern` there is not on a `@ConfigurationProperties` class, hence the
need to write the json file "by hand"
Hope that makes sense

On Wed, Oct 14, 2015 at 6:31 PM, Gary Russell <notifications@github.com>
wrote:

> In
> common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/file/remote/AbstractRemoteFileSourceProperties.java
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/42#discussion_r42017328>
> :
>
> > +import java.io.File;
> > +
> > +/**
> > + * @author David Turanski
> > + * @author Gary Russell
> > + *
> > + */
> > +public abstract class AbstractRemoteFileSourceProperties {
> > +
> > +	/**
> > +	 * The remote FTP directory, "/" by default.
> > +	 */
> > +	private String remoteDir = "/";
> > +
> > +	/**
> > +	 * Set to true to delete remote files after successful transfer, false by default.
>
> Oh - looks like we're using the META-INF ...
>
> https://github.com/spring-cloud/spring-cloud-stream-modules/pull/44/files
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-cloud/spring-cloud-stream-modules/pull/42/files#r42017328>
> .
>